util/mvmap/mvmap.go

```diff
@@ -170,3 +181,42 @@ func (m *MVMap) hash(key []byte) uint64 {
 	m.hashFunc.Write(key)
 	return m.hashFunc.Sum64()
 }
+
+// Len returns the number of values in th mv map, the number of keys may be less than Len
+// if the same key is put more than once.
+func (m *MVMap) Len() int {
+	return m.length
+}
+
+// Iterator is used to iterate the MVMap.
+type Iterator struct {
+	m        *MVMap
+	sliceCur int
+	entryCur int
+}
+
+// Next return the next key/value pair of the MVMap.
+// It returns nil, nil when there is no more entry to iterate.
+func (i *Iterator) Next() (key, value []byte) {
+	for {
+		if i.sliceCur >= len(i.m.entryStore.slices) {
+			return nil, nil
+		}
+		entrySlice := i.m.entryStore.slices[i.sliceCur]
+		if i.entryCur >= len(entrySlice) {
+			i.sliceCur++
+			i.entryCur = 0
```

Comment:
Only the first entry in the first slice represents null, then it starts from 1, other slice starts from 0.